src/Compilers/Core/Portable/MetadataReference/Metadata.cs

```diff
@@ -6,15 +6,34 @@
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
+    /// An Id that can be used to identify a metadata instance.  If two metadata instances 
+    /// have the same id then they are guaranteed to have the same content.  If two metadata
+    /// instances have different ids, then the contents may or may not be the same.  As such,
+    /// the id is useful as a key in a cache when a client wants to share data for a metadata
+    /// reference as long as it has not changed.
+    /// </summary>
+    public sealed class MetadataId
+    {
```

Comment:
Okay.  Still wish there was something that could be added to at least give the type instances some debuggability value but I understand the strong type argument.